src/Workspaces/VisualBasic/Portable/Indentation/VisualBasicIndentationService.Indenter.vb

```diff
@@ -26,35 +26,40 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Indentation
-                    Return indenter.GetIndentationOfLine(previousLine)
+                    ' if the comment is the only thing on a line, then preserve its indentation for the next line.
+                    Dim line = indenter.Text.Lines.GetLineFromPosition(trivia.FullSpan.Start)
+                    If line.GetFirstNonWhitespacePosition() = trivia.FullSpan.Start Then
+                        Return New IndentationResult(trivia.FullSpan.Start, 0)
+                    End If
                 End If
 
-                If trivia.Kind = SyntaxKind.LineContinuationTrivia Then
-                    Return GetIndentationBasedOnToken(indenter, GetTokenOnLeft(trivia), trivia)
-                End If
+                ' preserve the indentation of the comment trivia before a case statement
+                If trivia.Kind = SyntaxKind.CommentTrivia Then
+                    If trivia.Token.IsKind(SyntaxKind.CaseKeyword) AndAlso
+                       trivia.Token.Parent.IsKind(SyntaxKind.CaseStatement) Then
+                        Return New IndentationResult(trivia.SpanStart, 0)
+                    End If
 
-                ' Line ends in comment
-                If trivia.Kind = SyntaxKind.CommentTrivia Then ' Two cases a line ending comment or _ comment
-                    Dim firstTrivia As SyntaxTrivia = indenter.Tree.GetRoot(indenter.CancellationToken).FindTrivia(token.Span.End + 1)
-                    ' firstTrivia contains either an _ or a comment, this is the First trivia after the last Token on the line
-                    If firstTrivia.Kind = SyntaxKind.LineContinuationTrivia Then
-                        Return GetIndentationBasedOnToken(indenter, GetTokenOnLeft(firstTrivia), firstTrivia)
-                    Else
-                        ' This is we have just a comment
-                        Return GetIndentationBasedOnToken(indenter, GetTokenOnLeft(trivia), trivia)
+                    ' Line ends in comment
+                    ' Two cases a line ending comment or _ comment
+                    If tokenOpt.HasValue Then
+                        Dim firstTrivia As SyntaxTrivia = indenter.Tree.GetRoot(indenter.CancellationToken).FindTrivia(tokenOpt.Value.Span.End + 1)
+                        ' firstTrivia contains either an _ or a comment, this is the First trivia after the last Token on the line
+                        If firstTrivia.Kind = SyntaxKind.LineContinuationTrivia Then
+                            Return GetIndentationBasedOnToken(indenter, GetTokenOnLeft(firstTrivia), firstTrivia)
+                        Else
+                            ' This is we have just a comment
+                            Return GetIndentationBasedOnToken(indenter, GetTokenOnLeft(trivia), trivia)
+                        End If
                     End If
                 End If
```

Comment:
this part is preserved.  if a line ends with `_` or `_ ' comment`, we properly indent against the last token that precedes that trivia.